packages/shopify-cart/src/client/index.ts

```diff
@@ -26,6 +26,16 @@ import type {
 } from '../types/shopify.type';
 import type { CustomFragments } from '../graphql/fragments';
 
+/**
+ * @param {string} shopifyStorefrontAccessToken - required Shopify Storefront API access token
+ * @param {CustomFragments} customFragments - optional GraphQL fragments for data customization
+ * @param fetchClient - optional Fetch API-compatible request function
+ * @param {string} shopifyCustomEndpoint - optional Shopify custom API endpoint
+ * @param {string} shopifyShopId - optional Shopify shop id
+ * @param {LanguageCode} language - optional Shopify Language Code used for cart queries/responses. Defaults to `EN` if not supplied
+ * @param {CountryCode} country - optional Shopify Country Code used for cart queries/responses. Defaults to `ZZ` (unknown region) if not supplied
+ * @param {string} locale - optional Nacelle `locale` used to assign `NacelleCartMerchandise` the correct `nacelleEntryId`. Defaults to `en-US` if not supplied.
+ */
```

Comment:
The way that this is set up currently, we aren't getting in-editor feedback when iintializing the `cartClient`:

https://user-images.githubusercontent.com/5732000/206814853-a7c62672-51c6-4184-8cde-7d8907d5395d.mov

Feel free to ignore the duplicate `fetchClient` param in that video. If you're also seeing it too, that'd be good to know, but I suspect is has something to do with the most recent VSCode update.

What I recommend doing instead is moving the interface property docs into the interface, directly above the property:


```suggestion
```


```ts
export interface CreateClientParams {
  /** Required Shopify Storefront API access token */
  shopifyStorefrontAccessToken: string;

  /** Optional GraphQL fragments for data customization */
  customFragments?: CustomFragments;

  /** Optional Fetch API-compatible request function */
  fetchClient?: typeof fetch;

  /** Shopify custom API endpoint. Required if not supplying a `shopifyShopId`. */
  shopifyCustomEndpoint?: string;

  /** Shopify Shop ID. Required if not supplying a `shopifyCustomEndpoint`. */
  shopifyShopId?: string;

  /** Optional Shopify Language Code used for cart queries/responses. Defaults to `EN` if not supplied. */
  language?: LanguageCode;

  /** Optional Shopify Country Code used for cart queries/responses. Defaults to `ZZ` (unknown region) if not supplied. */
  country?: CountryCode;

  /** Optional Nacelle `locale` used to assign `NacelleCartMerchandise` the correct `nacelleEntryId`. Defaults to `en-US` if not supplied. */
  locale?: string;
}
```

This is the pattern that the generated types in `packages/shopify-cart/src/types/cart.type.ts` use.

Example:

https://user-images.githubusercontent.com/5732000/206816309-96adeea2-5e59-495f-aae7-f1d8af8b3c26.mov